src/Sulu/Bundle/ContactBundle/Controller/AbstractMediaController.php

```diff
@@ -135,10 +135,11 @@ public function __construct(
      * @param string $entityName
      * @param string $id
      * @param string $mediaId
+     * @param callable|null $dispatchDomainEvent
      *
      * @return Media
      */
-    protected function addMediaToEntity($entityName, $id, $mediaId)
+    protected function addMediaToEntity($entityName, $id, $mediaId, $dispatchDomainEvent = null)
```

Comment:
think it would be a bit easier to understand to pass the domain event and dispatch it here instead of passing a callable? 🙂

Reply:
you mean passing the `ContactDomainEvent::class` string? This would force every domain event passed here to have a constructor like `__construct($contact, $media)` ... I guess passing a callable is cleaner here, also the whole ContactBundle is built with callables like this one.

Reply:
no i thought that we can already create the event. but i figured that we cannot do that because we dont have the entities there, so this way is fine for me 🙂

Reply:
👍

Reply:
think i would call it `dispatchDomainEventCallback` then - guess you will need to rebase this pr anyway once #5990 is merged 🙂